### ISSUE TITLE
[ISSUE] Issue with `databricks_sql_table` resources with `storage_location` set with trailing slash

### ISSUE THREAD

**Issue by bradfordchang** [databricks/terraform-provider-databricks#2316]
### Configuration
```hcl
resource "databricks_sql_table" "table_name" {
  catalog_name = "catalog_name"
  schema_name = "schema_name"
  name = "table_name"
  table_type = "EXTERNAL"
  storage_location = "s3://bucket/path/"
  storage_credential_name = "credential_name"
}
```

### Expected Behavior

Subsequent deployments without other changes should show the `databricks_table` resource as unchanged.

### Actual Behavior

Terraform attempts to update in-place, and update the `storage_location`:

```
  # databricks_sql_table.table_name will be updated in-place
  ~ resource "databricks_sql_table" "table_name" {
        id                      = "catalog_name.schema_name.table_name"
        name                    = "table_name"
      ~ storage_location        = "s3://bucket/path" -> "s3://bucket/path/"
```

And fails with:

```
databricks_sql_table.table_name: Modifying... [id=catalog_name.schema_name.table_name]
╷
│ Error: cannot update sql table: cannot execute ALTER TABLE catalog_name.schema_name.table_name SET LOCATION 's3://bucket/path/' WITH (CREDENTIAL `credential_name`): [PARSE_SYNTAX_ERROR] Syntax error at or near 'WITH'.(line 1, pos 138)
│ 
│   with databricks_sql_table.table_name,
│   on main.tf line 33, in resource "databricks_sql_table" "table_name":
│   33: resource "databricks_sql_table" "table_name" {
│ 
╵
```

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply`

### Terraform and provider versions

```
Terraform v1.4.6
on darwin_arm64
+ provider registry.terraform.io/databricks/databricks v1.16.1
```

### Debug Output

[Debug Output](https://gist.github.com/bradfordchang/257f1b21670796985ca436281595e3b7)

### Important Factoids

In the meantime, this can be worked around by not putting trailing slashes on `storage_location`.

**Comment by bradfordchang** [comment]
Maybe we should strip all trailing slashes?

**Comment by abaiju15** [comment]
@alexott this same thing seems to be happening in external locations, catalogs, & schemas. We're seeing all these recreates in the plans while trying to add new catalogs & schemas to an existing configuration:

```terraform
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the
last "terraform apply":

  # module.naz_metastore_non_prod_data.databricks_external_location.external_locations["nazbtecheu2ndear.demand-estimation"] has been changed
  ~ resource "databricks_external_location" "external_locations" {
        id              = "lola-nazbtecheu2ndear-demand-estimation"
        name            = "lola-nazbtecheu2ndear-demand-estimation"
      ~ url             = "abfss://demand-estimation@nazbtecheu2ndear.dfs.core.windows.net" -> "abfss://demand-estimation@nazbtecheu2ndear.dfs.core.windows.net/"
        # (5 unchanged attributes hidden)
    }
  # module.naz_metastore_non_prod_data.databricks_catalog.catalogs["finance_dev"] has been changed
  ~ resource "databricks_catalog" "catalogs" {
        id            = "finance_dev"
        name          = "finance_dev"
      ~ storage_root  = "abfss://finance@nazbtecheu2nfinance.dfs.core.windows.net" -> "abfss://finance@nazbtecheu2nfinance.dfs.core.windows.net/"
        # (4 unchanged attributes hidden)
    }
... [there's a lot more here]
Plan: 42 to add, 19 to change, 14 to destroy.
```

**Comment by alexott** [comment]
Please update provider - it was fixed 2 weeks ago in #2336
